Makefile

```diff
@@ -1,20 +1,35 @@
 all: kops
 
-DOCKER_REGISTRY=gcr.io/must-override/
-S3_BUCKET=s3://must-override/
+DOCKER_REGISTRY?=gcr.io/must-override/
```

Comment:
What is the difference in a makefile between ?=, = and := ?

(I was going to ask you privately, but then I figured better to share with the world!)